concourse/steps/malware_scan.mako

```diff
@@ -65,35 +66,31 @@ filter_function = create_composite_filter_function(
   include_component_names=${filter_cfg.include_component_names()},
   exclude_component_names=${filter_cfg.exclude_component_names()},
 )
-
-image_references = [
-  resource.access.imageReference
-  for component, resource
-  in product.v2.enumerate_oci_resources(
-    component_descriptor=component_descriptor,
-  )
-  if filter_function(component, resource)
-]
-
+id = uuid.uuid4()
 ci.util.info('running malware scan for all container images')
 malware_scan_results = [
   scan_result
-  for scan_result in virus_scan_images(image_references, '${clam_av.clamav_cfg_name()}')
+  for scan_result in virus_scan_images(
+    component_descriptor,
+    filter_function,
+    '${clam_av.clamav_cfg_name()}',
+  )
 ]
 malware_scan_results_with_findings = [
   sr for sr in malware_scan_results
-  if sr.finding != 'No malware detected.'
+  if sr.findings
 ]
-ci.util.info(f'{len(image_references)} image(s) scanned for virus signatures.')
 print(
   tabulate.tabulate(
-    map(lambda dc: dataclasses.astuple(dc), malware_scan_results),
-    headers=MalwareScanResult.headers(),
+    map(lambda sr: (sr.resource.name, sr.scan_state,'\n'.join(sr.findings)), malware_scan_results),
```

Comment:
since you touch this line of code: maybe change to list-comprehension (or maybe even better a generator-expr)?